<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <include file="$(find hard_lecture)/launch/mavros.launch"/>
  <include file="$(find hard_lecture)/launch/ydlidar_x4.launch"/>
  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch"/>

  <node name="twist_publisher" pkg="nav_lecture" type="nav_twist_publisher" output="screen">
    <remap from="joy" to="/mavros_bridge/joy"/>
    <remap from="cmd_vel" to="/mavros/setpoint_velocity/cmd_vel"/>
    <remap from="activate" to="/mavros_bridge/activate"/>
    <param name="assign_x" value="0"/>
    <param name="assign_y" value="-1"/>
    <param name="assign_z" value="3"/>
    <param name="max_x" value="-0.5"/>
    <param name="max_z" value="-1.5"/>

    <param name="assign_activate" value="0"/>
  </node>

  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" clear_params="true">
    <rosparam command="load" file="$(find nav_lecture)/config/gmapping.yaml" />
    <!-- <remap from="scan" to="front_laser/scan" /> -->
    <param name="base_frame" value="base_link" />
    <param name="odom_frame" value="odom" />
    <param name="map_frame"  value="map" />
  </node>


  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser4"
    args="0.1 0.0 0.0 3.1415 0.0 0.0 /base_link /laser_frame 40" />

</launch>
